sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1813,6 +1817,7 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       val path = dir.getCanonicalPath
       primitiveFieldAndType
         .toDF("value")
+        .repartition(1)
```

Comment:
why is the `repartition` required?

Reply:
As far as I remember I added the `repartition(1)` here and in other places because to eliminate empty files. Such empty files are produced by empty partitions. Probably we could avoid writing empty files at least in the case of text-based datasources but any case let's look at `TextOutputWriter`, for example. It creates an input stream for a file in its constructor: https://github.com/apache/spark/blob/46110a589f4e91cd7605c5a2c34c3db6b2635830/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextFileFormat.scala#L151
and closes the empty file in https://github.com/apache/spark/blob/46110a589f4e91cd7605c5a2c34c3db6b2635830/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextFileFormat.scala#L162 . So, even if we didn't write anythings to the file, it creates an empty file.

From the read side, when `Jackson` parser tries to read the empty file, it cannot detect any JSON tokens on the root level and returns null from  `nextToken()` for which I throw a bad record exception for now -> `Row(...)` in `PERMISSIVE` mode.